packages/internal-test-env/index.ts

```diff
@@ -0,0 +1,351 @@
+  createSolidDataset,
+  deleteSolidDataset,
+  getPodUrlAll,
+  getSourceIri,
+  saveSolidDatasetInContainer,
+} from "@inrupt/solid-client";
+
+const availableEnvironment = [
+  "ESS Dev-Next" as const,
+  "ESS PodSpaces" as const,
+  "NSS" as const,
+  //  "ESS PodSpaces Next" as const,
+];
+
+export type AvailableEnvironment = typeof availableEnvironment extends Array<
+  infer E
+>
+  ? E
+  : never;
+
+const availableProtocol = ["ESS Notifications Protocol" as const];
+
+export type AvailableProtocol = typeof availableProtocol extends Array<infer E>
+  ? E
+  : never;
+
+export interface TestingEnvironmentNode {
+  environment: AvailableEnvironment;
+  idp: string;
+  notificationGateway: string | undefined;
+  protocol: AvailableProtocol | undefined;
+  clientCredentials: {
+    requestor: {
+      id: string;
+      secret: string;
+    };
+    resourceOwner: {
+      id: string | undefined;
+      secret: string | undefined;
+    };
+  };
+  vcProvider: string | undefined;
+  features: FeatureFlags | undefined;
+}
+export interface TestingEnvironmentBrowser {
+  clientCredentials: {
+    resourceOwner: {
+      login: string;
+      password: string;
+    };
+  };
+  idp: string;
+  notificationGateway: string | undefined;
+  features: FeatureFlags | undefined;
+}
+
+type FeatureFlags = {
+  [key: string]: any;
+};
+
+export interface EnvVariables {
+  // Common Envs
+  E2E_TEST_ENVIRONMENT: AvailableEnvironment;
+  E2E_TEST_IDP: string;
+  E2E_TEST_USER: string;
+  E2E_TEST_PASSWORD: string;
+
+  // Client credentials for the access requestor
+  E2E_TEST_REQUESTOR_CLIENT_ID: string;
+  E2E_TEST_REQUESTOR_CLIENT_SECRET: string;
+
+  // Needed for solid-notifications-js
+  E2E_TEST_NOTIFICATION_GATEWAY: string | undefined;
+  E2E_TEST_NOTIFICATION_PROTOCOL: AvailableProtocol | undefined;
+
+  // VC service provider
+  E2E_TEST_VC_PROVIDER: string | undefined;
+
+  // Client credentials for the resource owner
+  E2E_TEST_RESPONDER_CLIENT_ID: string | undefined;
+  E2E_TEST_RESPONDER_CLIENT_SECRET: string | undefined;
+
+  E2E_TEST_FEATURE_ACP: boolean | undefined;
+  E2E_TEST_FEATURE_ACP_V3: boolean | undefined;
+  E2E_TEST_FEATURE_WAC: boolean | undefined;
+}
+
+let envLoaded = false;
+let featuredFlags: FeatureFlags = {};
+
+export function setupEnv() {
+  // If we're in CI, the environment is already configured.
+  if (process.env.CI) {
+    return;
+  }
+
+  if (envLoaded) {
+    return;
+  }
+
+  const envPath = join(process.cwd(), "e2e/env");
+
+  // Otherwise load dotenv configuration
+  config({
+    path: envPath,
+    silent: true,
+  });
+
+  if (!process.env.E2E_TEST_ENVIRONMENT) {
+    console.error(
+      `We didn't find the given environment variable E2E_TEST_ENVIRONMENT,
+tried looking in the following directory for \`.env.local \`: 
+${envPath}`
+    );
+  }
+
+  // Creating feature flag object if there are feature flags
+  Object.keys(process.env)
+    .filter((envVar) => /E2E_TEST_FEATURE.*/.test(envVar))
```

Comment:
```suggestion
    .filter((envVar) => envVar.startsWith("E2E_TEST_FEATURE_"))
```

No need for regex here.